src/main/java/de/sldk/mc/metrics/TickDurationCollector.java

```diff
@@ -0,0 +1,54 @@
+package de.sldk.mc.metrics;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.logging.Level;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Server;
+import org.bukkit.plugin.Plugin;
+
+import io.prometheus.client.Gauge;
+
+public abstract class TickDurationCollector extends Metric {
+    /*
+     * If reflection is successful, this will hold a reference directly to the
+     * MinecraftServer internal tick duration tracker
+     */
+    private static long[] tickDurationReference = null;
+
+    public TickDurationCollector(Plugin plugin, Gauge gauge, String name) {
+        super(plugin, gauge);
+
+        /*
+         * If there is not yet a handle to the internal tick duration buffer, try
+         * to acquire one using reflection.
+         *
+         * This searches for any long[] array in the MinecraftServer class. It should
+         * work across many versions of Spigot/Paper and various obfuscation mappings
+         */
+        if (tickDurationReference == null) {
+            try {
+                /* Get the actual minecraft server class */
+                Server server = Bukkit.getServer();
+                Method getServerMethod = server.getClass().getMethod("getServer");
+                Object minecraftServer = getServerMethod.invoke(server);
+
+                /* Look for the only array of longs in that class, which is tick duration */
+                for (Field field : minecraftServer.getClass().getSuperclass().getDeclaredFields()) {
```

Comment:
I think we should make sure here that it's actually the **only** array of longs? Currently you're using the *first* one that's available. Basically, if we should ever encounter multiple candidates -> fail with an error and hope that someone reports it so that we can think of some special handling for that. What do you think?

Reply:
How about using the longest array of longs? Worst case you just get garbage data, it won't cause a crash or anything since you're still reading from an array of numbers either way.